src/snowflake/cli/api/metrics.py

```diff
@@ -86,7 +207,109 @@ def increment_counter(self, name: str, value: int = 1) -> None:
         else:
             self._counters[name] += value
 
+    @property
+    def _current_step(self) -> Optional[_CLIMetricsStep]:
+        return self._executing_steps[-1] if len(self._executing_steps) > 0 else None
+
+    @contextmanager
+    def track_step(self, name: str):
```

Comment:
let's workshop the name

Reply:
AFAIK the common nomenclature is "trace", which are composed of "spans". Both datadog and opentelemetry use this: https://opentelemetry.io/docs/concepts/signals/traces/

Reply:
I chose "start_span" as the new name based on my research below:

Every library seems to yield some kind of `Span` object, but in terms of function names, datadog uses `.trace()`, otel uses `.start_as_current_span()`, and sentry uses `.start_span()`, so I went with the simplest common denominator which to me was `start_span()`, since I guess we would count our instance of `CLIMetrics` as a trace in our context. I'm also open to renaming that class if we feel the nomenclature does not fit the concept